geopy/geocoders/googleplaces.py

```diff
@@ -0,0 +1,210 @@
+            more information, see documentation on
+            :class:`urllib2.ProxyHandler`.
+        """
+        super(GooglePlaces, self).__init__(
+            scheme=scheme, timeout=timeout, proxies=proxies, user_agent=user_agent
+        )
+
+        self.api_key = api_key
+        domain = 'maps.googleapis.com'
+
+        self.autocomplete_api = '{scheme}://{domain}/maps/api/place/autocomplete/json'.format(scheme=self.scheme, domain=domain)
+        self.details_api = '{scheme}://{domain}/maps/api/place/details/json'.format(scheme=self.scheme, domain=domain)
+
+    @staticmethod
+    def _format_components_param(components):
+        """
+        Format the components dict to something Google understands.
+        """
+        return "|".join(
+            (":".join(item)
+             for item in components.items()
+             )
+        )
+
+    def geocode(self,
+                query,
+                exactly_one=False,
+                timeout=None,
+                location=None,
+                radius=None,
+                language=None,
+                types=None,
+                components=None,
+                ):  # pylint: disable=W0221,R0913
+        """
+        Geocode a location query.
+
+        :param string query: The address or query you wish to geocode.
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+
+            .. versionadded:: 1.12
+
+        :type location: :class:`geopy.point.Point`, list or tuple of (latitude,
+            longitude), or string as "%(latitude)s, %(longitude)s"
+
+        :param int radius:  The distance (in meters) within which to return place results.
+            Setting a radius biases results to the indicated area, but may not fully restrict results to the specified area.
+
+        :param string language: The language in which to return results. List of supported languages: https://developers.google.com/maps/faq#languagesupport
+
+        :param string types: The types of place results to return.
+            Possible values are "geocode", "address", "establishment", "(regions)" or "(cities)"
+            If no type is specified, all types will be returned.
+
+        :param dict components: A grouping of places to which you would like to restrict your results.
+            Currently, you can use components to filter by country.
+            For example: components=country:fr would restrict your results to places within France.
+        """
+        autocomplete_predictions = self.autocomplete(query, timeout, location, radius, language, types, components)
+        if autocomplete_predictions and exactly_one:
```

Comment:
according to the documentation form geopy if exactly_one is true we should return a single object, not a list of length one. (this is how the googlev3 geocoder works).

it seems rather stupid to me, but that's how geopy works and unless we really want to fork it (which i don't) we should follow its api.